### PR TITLE
chore: Prevent redundant Sentry sourcemap uploads

### LINKED ISSUE #10812: Prevent redundant Sentry sourcemap uploads
### What is this about?

This task tracks preventing redundant Sentry sourcemap uploads. For context: We've noticed that there are duplicate sourcemaps being uploaded for releases under the same distribution name. This messes with stack traces since production bugs can now be associated with sourcemaps generated with different stack traces. In turn, this can make debugging Sentry issues impossible. The solution is to fix the system so that we only upload one set of source maps to each platform for each distributed version. 

This shows redundant sourcemaps uploaded to the same distribution name
image

### Scenario

_No response_

### Design

_No response_

### Technical Details

- It is believed that the issue occurs because certain release flows are used indiscriminately
- For example, `build_ios_release`/`build_android_release` is a step used for cutting RCs as well as uploading builds to TestFlight for testing 
  - This pipeline should be renamed/created to describe the building and staging of apps in test disruptions
  - `release_builds_to_store_pipeline` - remove this pipeline
  -  `build_send_apps_to_testflight` - create a pipeline that disables source map uploads
- In the scenario above, it's possible that `build_ios_release`/`build_android_release` overwrites the sourcemaps for the RC, which eventually becomes prod
- The technical task here is to confirm this suspicion and possibly abstract + designate a flow only used for cutting RCs. Once this flow is created, we need to ensure that only this flow uploads Sentry sourcemaps

### Threat Modeling Framework

_No response_

### Acceptance Criteria

- Only a single set of sourcemaps is being uploaded to each RC/distribution version
- Reported errors in Sentry shows the correct stack trace
- The new abstracted flow creates an RC + uploads the correct sourcemaps in Sentry

### Stakeholder review needed before the work gets merged

- [ ] Engineering (needed in most cases)
- [ ] Design
- [ ] Product
- [ ] QA (automation tests are required to pass before merging PRs but not all changes are covered by automation tests - please review if QA is needed beyond automation tests)
- [ ] Security
- [ ] Legal
- [ ] Marketing
- [ ] Management (please specify)
- [ ] Other (please specify)

### References

_No response_

### PR DESCRIPTION
**Pull request by Daniel-Cross**
## **Description**

Made a change to `bitrise.yml` that renames the flow for builds that are made specifically for testing purposes and not for production.

The change adds an env variable that skips the creation of Sentry source map files if the env variable is found.

## **Related issues**

Fixes:

## **Manual testing steps**

1. Trigger a build that doesn't say source maps -> see source map upload is skipped
2. Trigger a build that does say source maps -> see source map upload is completed
3.

### **Before**

NA

### **After**

`build_android_release_with_sourcemaps`
Screenshot 2024-09-24 at 8 58 27 AM

`build_android_release`
Screenshot 2024-09-24 at 8 59 28 AM

`build_ios_release_with_sourcemaps`

`build_ios_release`
Screenshot 2024-09-24 at 9 02 04 AM

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: bitrise.yml [hunk 1]
```diff
@@ -1054,10 +1057,11 @@ workflows:
       - deploy-to-bitrise-io@2.2.3:
           is_always_run: false
           is_skippable: true
+          run_if: '{{enveq "SKIP_SENTRY_UPLOAD" ""}}'
           inputs:
             - pipeline_intermediate_files: sourcemaps/android/index.js.map:BITRISE_PLAY_STORE_SOURCEMAP_PATH
             - deploy_path: sourcemaps/android/index.js.map
-          title: Bitrise Deploy Sourcemaps
+          title: Deploy Source Map
```
Comment (sethkfman) [inline bitrise.yml:1064]: This deploy source map only adds the artifacts to Bitrise storage. You will need to modify the Sentry source map generation in the `build.sh` file

Comment (Daniel-Cross) [reply bitrise.yml:1060]: Changed the variable in `build.sh` to address this issue

File: scripts/build.sh [hunk 2]
```diff
@@ -292,7 +292,7 @@ buildIosRelease(){
   	remapEnvVariableRelease
 
 	# Enable Sentry to auto upload source maps and debug symbols
```
Comment (legobeat) [inline scripts/build.sh:294]:
```suggestion
	# Disable Sentry auto-upload of source maps and debug symbols
```

File: scripts/build.sh [hunk 3]
```diff
@@ -291,8 +291,8 @@ generateArchivePackages() {
 buildIosRelease(){
   	remapEnvVariableRelease
 
-	# Enable Sentry to auto upload source maps and debug symbols
-	export SENTRY_DISABLE_AUTO_UPLOAD="false"
+	# Disable Sentry auto-upload of source maps and debug symbols
+	export SENTRY_DISABLE_AUTO_UPLOAD="true"
```
Comment (sethkfman) [inline scripts/build.sh:295]: We will also need to do this for Android. I think can be done on line 418 for Android.

File: bitrise.yml [hunk 4]
```diff
@@ -89,11 +89,11 @@ stages:
       - build_ios_release: {}
       - build_android_qa: {}
       - build_ios_qa: {}
-  create_build_release:
+  create_build_for_test_flight:
     workflows:
       - build_android_release: {}
```
Comment (sethkfman) [inline bitrise.yml:94]: I think easiest and most explicit way to do this it chain the workflows and Runway can used the explicit workflow. Something like this:
```
  build_ios_release_with_sourcemaps:
    envs:
      - NO_FLIPPER: '1'
      - SENTRY_DISABLE_AUTO_UPLOAD: false
    after_run:
      - build_ios_release
```

File: bitrise.yml [hunk 5]
```diff
@@ -85,15 +85,20 @@ pipelines:
...
+      - SENTRY_DISABLE_AUTO_UPLOAD: false
+    workflows:
+      # - build_android_release: {}
       - build_ios_release: {}
-      - build_android_qa: {}
-      - build_ios_qa: {}
-  create_build_release:
+      # - build_android_qa: {}
+      # - build_ios_qa: {}
+  create_build_for_test_flight:
```
Comment (Daniel-Cross) [inline bitrise.yml:97]: @sethkfman Added this block here based on your comment.

File: bitrise.yml [hunk 6]
```diff
@@ -89,14 +89,24 @@ stages:
...
     workflows:
       - build_android_release: {}
       - build_ios_release: {}
-  deploy_build_release:
+  deploy_build_for_test_flight:
     workflows:
       - deploy_android_to_store: {}
       - deploy_ios_to_store: {}
+  build_ios_release_with_sourcemaps:
```
Comment (sethkfman) [inline bitrise.yml:100]: @Daniel-Cross these items are in the `stages` area. Can you move it down to the `workflows`?

**Review by sethkfman — APPROVED** [review]
LGTM

**Pull request merged**